SalesforceFoundation/EDA
# [EDA] When deleting an address, the filter field and usernames to exclude filters are not working in the trigger handler for the address trigger ADDR_Addresses_TDTM.

# Critical Changes

# Changes
## General Updates
- We've updated EDA to ensure that when an Address record that matches the ADDR_Addresses_TDTM trigger handler record filter is deleted, the Contact's mailing address fields retain their values.

# Issues Closed
Closes #988

# New Metadata
ADDR_Addresses_TEST

# Deleted Metadata

# Testing Notes
1. Delete an Address record that matches the ADDR_Addresses_TDTM trigger handler record filter. 
- Create a Contact
- Create an Address record from Contact's Address related list and set the "Default Address" checkbox to true. **Add a value on "Mailing City" that you will want to add to the filter value on ADDR_Addresses_TDTM trigger handler record. 
- Result: The Contact's mailing address fields and "Current Address" lookup field should be auto populated. 
- Update the ADDR_Addresses_TDTM trigger handler record and add a filter field + value. ie. Set the filter field = MailingCity__c, and set the filter value = 'Boston' and save the changes. 
- Go to the Address record created from the Contact
- Delete the Address record
- Result: The Contact's mailing address fields should **NOT** be blank. The "Current Address" lookup field on the Contact should be blank. The Address record is deleted. 
2. Delete an Address record that does not match the  ADDR_Addresses_TDTM trigger handler record filter. 
- Create a Contact
- Create an Address record from Contact's Address related list and set the "Default Address" checkbox to true. **Ensure that the "Mailing City" on the Address does not match the filter value set in the ADDR_Addresses_TDTM trigger handler record. 
- Result: The Contact's mailing address fields and "Current Address" lookup field should be auto populated. 
- Update the ADDR_Addresses_TDTM trigger handler record and add a filter field + value. ie. Set the filter field = MailingCity__c, and set the filter value = 'Boston' and save the changes. 
- Go to the Address record created from the Contact
- Delete the Address record
- Result: The Contact's mailing address fields should be blank. The "Current Address" lookup field on the Contact should be blank. The Address record is deleted.
